### PR TITLE
[REFACTOR][OBJECT] Consoldiate NodePtr/Ref/Hash/Equal  to Object

### PR DESCRIPTION
**Pull request by tqchen**
Historically, we have classes like NodePtr/Ref/HashEqual.
After unified object protocol, these names are just alias of the object counterpart.
Moreover, there are helper macros defined over the places for defining these object.

This PR consoldiate the terminologies into the corresponding ones
in the Object system so we have a clean and consistent API moving forward.

## Changes

NodePtr/Ref/Hash/Equal -> ObjectPtr/Ref/Hash/Equal
TVM_DEFINE_NODE_REF_METHODS -> TVM_DEFINE_OBJECT_REF_METHODS
consoldate relay macros into the object ones

### CODE REVIEW

**Comment by tqchen** [comment]
cc @jroesch @wweic @yzhliu @Hzfengsy @ZihengJiang

File: include/tvm/attrs.h [hunk 1]
```diff
@@ -85,7 +85,7 @@ namespace tvm {
  */
 template<typename TNodeRef>
```
Comment (wweic) [inline include/tvm/attrs.h]:
```suggestion
template<typename TObjectRef>
```

**Comment by merrymercy** [comment]
Do we also want to do renaming in python code? There are still NodeBase, register_node in python

**Comment by merrymercy** [comment]
@tqchen  Do we also want to do renaming in python code? There are still NodeBase, register_node in python.

Given a lot of rename has been done recently in several PRs. Could you also provide a upgrade instructions for us to update our codebase.

**Comment by tqchen** [comment]
@merrymercy yes we should update the python side as well, please feel free to open PRs :)

Here is the guide #4647 , please feel free to add comments

**Pull request merged**